Automattic/jetpack
# Sync: IDC: Add get_raw_url method to Jetpack_Sync_Functions

For sites that use domain mapping and multi language, it can be quite hard to sync uniform URLs back to WP.com, which can cause the site to be put in IDC.

To address this, I'd like to start syncing the unfiltered values for `home` and `siteurl`. This will introduced some jankiness where we sync the non-mapped domain for domain mapped sites. But, we can then address that by adding support for the top multi language and domain mapping plugins.

I've added support for Donncha's and WPMU Dev's domain mapping plugins. But, it's untested at the moment.

To test:

- Create network and point domains to network
- Checkout branch
- Install and configure one of the domain mapping plugins
- Ensure that the mapped domain gets sent to WPCOM
- You can comment out [this line](https://github.com/Automattic/jetpack/pull/5852/files#diff-2f6072543fb2336fcfb4dcbb3949b0f3R113) to make sure that the non-mapped domain gets synced properly as well (this simulates the raw URL getting synced when a non-supported domain mapping plugin is used
- Repeat steps for other domain mapping plugin
- Ensure there are no errors


Note: it could take several minutes the URLs to be synced each time you change.
